tests/qos/test_buffer.py

```diff
@@ -0,0 +1,981 @@
+    if not default_mtu:
+        logging.info("Default MTU {}".format(default_mtu))
+        default_mtu = duthost.shell('redis-cli -n 4 hget "PORT|{}" mtu'.format(port_to_test))['stdout']
+
+
+def load_lossless_headroom_data(duthost):
+    """Load test parameters from the json file. Called only once when the module is initialized
+
+    Args:
+        duthost: the DUT host object
+    """
+    global default_lossless_headroom_data
+    if not default_lossless_headroom_data:
+        dut_hwsku = duthost.facts["hwsku"]
+        dut_platform = duthost.facts["platform"]
+        skudir = "/usr/share/sonic/device/{}/{}/".format(dut_platform, dut_hwsku)
+        lines = duthost.shell('cat {}/pg_profile_lookup.ini'.format(skudir))["stdout"]
+        default_lossless_headroom_data = {}
+        for line in lines.split('\n'):
+            if line[0] == '#':
+                continue
+            tokens = line.split()
+            speed = tokens[0]
+            cable_length = tokens[1]
+            size = tokens[2]
+            xon = tokens[3]
+            xoff = tokens[4]
+            if not default_lossless_headroom_data.get(speed):
+                default_lossless_headroom_data[speed] = {}
+            default_lossless_headroom_data[speed][cable_length] = {'size': size, 'xon': xon, 'xoff': xoff}
+        default_lossless_headroom_data = default_lossless_headroom_data
+
+
+def load_test_parameters(duthost):
+    """Load test parameters from the json file. Called only once when the module is initialized
+
+    Args:
+        duthost: The DUT host object
+    """
+    global default_cable_length_list
+    global testparam_headroom_override
+    global testparam_lossless_pg
+
+    param_file_name = "qos/files/dynamic_buffer_param.json"
+    with open(param_file_name) as file:
+        params = json.load(file)
+        logging.info("Loaded test parameters {} from {}".format(params, param_file_name))
+        asic_type = duthost.facts['asic_type']
+        vendor_specific_param = params[asic_type]
+        default_cable_length_list = vendor_specific_param['default_cable_length']
+        testparam_headroom_override = vendor_specific_param['headroom-override']
+        testparam_lossless_pg = vendor_specific_param['lossless_pg']
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_module(duthost):
+    """Set up module. Called only once when the module is initialized
+
+    Args:
+        duthost: The DUT host object
+    """
+    detect_buffer_model(duthost)
+    if buffer_model_dynamic:
+        detect_ingress_pool_number(duthost)
+        load_lossless_headroom_data(duthost)
+        load_test_parameters(duthost)
+
+        logging.info("Cable length: default {}".format(default_cable_length_list))
+        logging.info("Ingress pool number {}".format(default_ingress_pool_number))
+        logging.info("Lossless headroom data {}".format(default_lossless_headroom_data))
+    else:
+        pytest.skip("Dynamic buffer isn't enabled, skip the test")
```

Comment:
Will this take care of skipping all the tests in this file if dynamic mode isn't enabled?

Reply:
Yes. That's the motivation.